src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,52 @@
+package duke.task;
+
+/**
+ * Represents a task with a deadline.
+ */
+public class Deadline extends Task {
+    private TaskDateTime dueTime;
+
+    /**
+     * Creates a new deadline with the given description and due time.
+     *
+     * @param description The description of the deadline.
+     * @param due         The due time of the deadline.
+     * @throws DukeDateTimeParseException If the due time is not in the correct
+     *                                    format.
+     */
+    public Deadline(String description, String due) throws DukeDateTimeParseException {
```

Comment:
Perhaps DukeDateTimeParseException is a little lengthy, could possible omit Duke and just put DateTimeParseException?

Reply:
I would love to do that, but there is an Exception class in java with the exact same name 😄